dask/dataframe/utils.py

```diff
@@ -366,6 +370,25 @@ def make_meta_object(x, index=None):
     raise TypeError("Don't know how to create metadata from {0}".format(x))
 
 
+def make_meta_util(x, index=None, parent_meta=None):
+    import dask.dataframe as dd
+
+    if isinstance(x, (dd.core.Series, dd.core.DataFrame)):
```

Comment:
I think `dd.core.Series` and `dd.core.DataFrame` already have `_meta`, no? In that case this may be redundant and could be removed in favor of the condition immediately below: `if hasattr(x, "_meta")`.

Reply:
Right, `make_meta_object` does something like the following to capture these cases:

```python
    if hasattr(x, "_meta"):
        return x._meta
    elif is_arraylike(x) and x.shape:
        return x[:0]
```

However, if that logic is already captured in `make_meta_object` (which is now registered with `make_meta_obj`), it doesn't seem like this check is necessary at all.  Is this correct?

Reply:
Good catch. I think your assessment is right, Rick. Unless I'm missing something, we could indeed remove both checks here.

Reply:
Yes, this is a redundant check. I got rid of it in `make_meta_object`. The reason being since `make_meta` dispatch was registered against generic python `object` type, all objects would simply pass through the `make_meta`, but with the change being made this won't happen so it will be the task of `make_meta_util` to handle this. The `is_arraylike` check however shouldn't be done in `make_meta_util` as objects like `pd.Series`/`cudf.Series` need to go through the dispatch mechanism below.